packages/react-query/src/useSuspenseQuery.test-d.ts

```diff
@@ -0,0 +1,71 @@
+import { expectError, expectType } from 'tsd'
+import { useSuspenseQuery } from '../dist'
+
+const queryKey = ['key'] as const
+const queryFn = async () => 'response' as const
+const boolean = Math.random() > 0.5
```

Comment:
I think it would be good to declare it like this, so how about fix the `useInfiniteQuery` part separately as boolean?